source/common/http/filter_manager.cc

```diff
@@ -573,8 +573,14 @@ void FilterManager::decodeHeaders(ActiveStreamDecoderFilter* filter, RequestHead
     ASSERT(!(state_.filter_call_state_ & FilterCallState::DecodeHeaders));
     state_.filter_call_state_ |= FilterCallState::DecodeHeaders;
     (*entry)->end_stream_ = (end_stream && continue_data_entry == decoder_filters_.end());
+    if ((*entry)->end_stream_) {
+      state_.filter_call_state_ |= FilterCallState::LastDataFrame;
+    }
     FilterHeadersStatus status = (*entry)->decodeHeaders(headers, (*entry)->end_stream_);
     state_.filter_call_state_ &= ~FilterCallState::DecodeHeaders;
+    if ((*entry)->end_stream_) {
+      state_.filter_call_state_ &= ~FilterCallState::LastDataFrame;
```

Comment:
hm, I see why you had to do this and it looks like  safe change, but I think you need to update the comment and probably rename for clarity.
/wait

Reply:
@alyssawilk I'm guessing you're referring to the comment and name of `LastDataFrame`, but if you're not, please let me know.
Proposed change:
```
diff --git a/source/common/http/filter_manager.h b/source/common/http/filter_manager.h
index e859b5bd45..9891cd540e 100644
--- a/source/common/http/filter_manager.h
+++ b/source/common/http/filter_manager.h
@@ -1056,9 +1056,11 @@ private:
       // to verify we do not encode1xx headers more than once per
       // filter.
       static constexpr uint32_t Encode1xxHeaders  = 0x100;
-      // Used to indicate that we're processing the final [En|De]codeData frame,
-      // i.e. end_stream = true
-      static constexpr uint32_t LastDataFrame = 0x200;
+      // Used to indicate that one of the following conditions is true:
+      // 1. The filter manager is processing the final [En|De]codeData frame.
+      // 2. The filter manager is in [En|De]codeHeaders and is processing a
+      //    header-only request or response.
+      static constexpr uint32_t EndOfStream = 0x200;
 
       // Masks for filter call state.
       static constexpr uint32_t IsDecodingMask = DecodeHeaders | DecodeData | DecodeMetadata | DecodeTrailers;
```
Does this seem sufficiently clear, or do you prefer a different name?